src/main/java/rs117/hd/HdPlugin.java

```diff
@@ -686,13 +693,34 @@ private void initPrograms() throws ShaderException
 			{
 				case "version_header":
 					return versionHeader;
-				case "MAX_MATERIALS":
-					return String.format("#define %s %d\n", key, MAX_MATERIALS);
-				case "CONST_MACOS_INTEL_WORKAROUND":
-					boolean isAppleM1 = OSType.getOSType() == OSType.MacOS && System.getProperty("os.arch").equals("aarch64");
-					return String.format("#define %s %d\n", key, config.macosIntelWorkaround() && !isAppleM1 ? 1 : 0);
-				case "MACOS_INTEL_WORKAROUND_MATERIAL_CASES":
-					return "return " + generateFetchMaterialCases(0, MAX_MATERIALS) + ";";
+				case "MATERIAL_CONSTANTS":
+				{
+					StringBuilder include = new StringBuilder();
+					for (Material m : Material.values())
+					{
+						include
+							.append("#define MAT_")
+							.append(m.name().toUpperCase())
+							.append(" getMaterial(")
+							.append(m.ordinal())
+							.append(")\n");
+					}
+					return include.toString();
+				}
+				case "MATERIAL_COUNT":
+					return String.format("#define %s %d", key, Material.values().length);
+				case "MATERIAL_GETTER":
+					return generateGetter("Material", Material.values().length);
+				case "WATER_TYPE_COUNT":
+					return String.format("#define %s %d", key, WaterType.values().length);
+				case "WATER_TYPE_GETTER":
+					return generateGetter("WaterType", WaterType.values().length);
+				case "LIGHT_COUNT":
+					return String.format("#define %s %d", key, Math.max(1, configMaxDynamicLights));
```

Comment:
Doesn't the `Math.max` here always limit the `LIGHT_COUNT` to 1? Or is light count just poorly named?

Reply:
Oh wait I'm thinking of `Math.min` nevermind :)